datalad/config.py

```diff
@@ -74,12 +80,35 @@ def get_git_version(runner=None):
                       protocol=StdOutErrCapture)['stdout'].split()[2]
 
 
-def _where_reload(obj):
+def _scope_reload(obj):
     """Helper decorator to simplify providing repetitive docstring"""
-    obj.__doc__ = obj.__doc__ % _where_reload_doc
+    obj.__doc__ = obj.__doc__ % _scope_reload_doc
     return obj
 
 
+#
+# TODO: remove in/after 0.16.0
```

Comment:
This is way too short.  Superficially looking through code on my laptop I see usage in -metalad, -container, -crawler, -webapp, -ukbiobank, -xnat, -hirni,  -osf, -mihextra, -neuroimaging, -deprecated. Plus hits in internal code, including individual analyses implementations. Given the breadth, I think we must assume that we break most non-trivial code somewhere when we stop supporting `where=`. I'd say 2-3 years is a more reasonable *minimum* threshold.

Realistically speaking, we can expect client library/tool code to change over time. But archived one-off implementations will be unlikely to be adjusted. This is of course not different from any other deprecation, but it touches such a core piece of code that the impact is more substantial in comparison. What pains me most is that any code that uses the metadata system (where config must be altered in `.datalad/config`) will be broken.

Reply:
```suggestion
# TODO: remove when deprecated 'where' is removed
```

Reply:
agree -- deprecation->removal should indeed be more "temporal" and not semver based!